pkg/pillar/devicenetwork/dns.go

```diff
@@ -150,21 +154,27 @@ func ResolveWithPortsLambda(domain string,
 						errs = append(errs, err)
 					}
 					if response != nil {
-						resolvedIPsChan <- response
+						select {
+						case <-quit:
+						case resolvedIPsChan <- response:
+						}
 					}
-					<-work
-					wg.Done()
 				}(dnsIPCopy, srcIPCopy)
 			}
 		}
 	}
 
 	wgChan := make(chan struct{})
+
 	go func() {
 		wg.Wait()
 		close(wgChan)
 	}()
 
+	defer func() {
```

Comment:
`resolvedIPsChan` and `work` channels are not closed at all. We could have a defer func to close them too.

Reply:
They get gc-ed, but  I added it explicitly.